docs/concept/allocation.qmd

```diff
@@ -237,24 +260,192 @@ Demand objectives come in 2 categories:
+
+With this we formulate the constraints
+
+$$
+	F_{(b_\text{upstream}, v)} = \sum_{p} F^p_v, \quad \forall v
+	\in FD.
+$$
+
+We define relative error variables per demand node per priority for which they have a demand:
+
+$$
+	E^p_v \ge 0, \quad \forall v \in (UD \cup FD),\\; p \in P_v.
+$$
+
+The constraints on these relative error terms are similar as before:
+
+$$
+	d_v^p \cdot E_v^p \ge d_v^p - F_v^p, \quad \forall v \in (UD \cup FD),\\; p \in P_v.
+$$
+
+With these relative error terms we formulate an objective per demand priority for which any of these nodes have a demand ($P^\text{flow} = \bigcup_{v \in UD \cup FD} P_v$):
+
+$$
+	\min \sum_{v,p} d_v^p \cdot E_v^p, \quad \forall p \in P^\text{flow}.
+$$
+
+Note the multiplication by the demand here, which makes this effectively a minimization in the absolute error. There is a reason we formulate relative in stead of absolute errors: to optimize for a fair distribution of water. To that end we can formulate per demand priority a global relative allocation error:
+
+$$
+	G^p = \frac{\sum_{v,p} d_v^p \cdot E_v^p}{\sum_{v,p} d^p_v}, \quad \forall p \in P^\text{flow},
+$$
+
+where in practice we multiply both sides by the denominator to prevent division by zero problems. Also note that the enumerator is precisely the expression for the aforementioned objective.
+Now in a subsequent objective, we want to penalize relative errors being larger than $G^p$. So we define new errors
+
+$$
+	\overline{E}_v^p \ge 0, \quad \forall v \in (UD \cup FD),\\; p \in P_v,
+$$
+
+on which we define the constraints
+
+$$
+	\overline{E}_v^p \ge E_v^p - G^p, \quad \forall v \in (UD \cup FD),\\; p \in P_v
+$$
+
+and with which we formulate the objectives
+
+$$
+	\min \sum_{v,p} \overline{E}_v^p, \quad \forall p \in P^\text{flow}.
+$$
+
+Of course the order of objectives is important. For each demand priority of the flow type described here, first the weighted sum of the errors $E_v^p$ must be minimized, then the sum of the errors $\overline{E}_v^p$.
+
+#### LevelDemand
+
+Say we have basins with LevelDemand $B$ where each basin $b \in B$ has a minimum level $h_{b,\min}^p$ for demand priorities $P^\min_b$ and a maximum level $h_{b,\max}^p$ for demand priorities $P_b^\max$. Furthermore each basin has a level to storage function $s_b(h)$. For all basins $b$ in the subnetwork (not just the ones with a level demand) we also have
+
+- The known initial level at the beginning of the allocation timestep $h_b^\text{init}$
+- The decision variable of the change in storage over the allocation timestep $\Delta S_b$
+
+For basins with a level demand $b \in B$ we additionally have
```

Comment:
```suggestion
Say we have Basins with LevelDemand $B$ where each Basin $b \in B$ has a minimum level $h_{b,\min}^p$ for demand priorities $P^\min_b$ and a maximum level $h_{b,\max}^p$ for demand priorities $P_b^\max$. Furthermore each Basin has a level to storage function $s_b(h)$. For all Basins $b$ in the subnetwork (not just the ones with a level demand) we also have

- The known initial level at the beginning of the allocation timestep $h_b^\text{init}$
- The decision variable of the change in storage over the allocation timestep $\Delta S_b$

For Basins with a level demand $b \in B$ we additionally have
```

I like to capitalize Basin here to make it clear we refer to the Basin node, not the hydrological concept of a basin.